### PR TITLE
Newsfeed functionality

### PR DESCRIPTION
**Pull request by LeonelHays**
## Description
https://trello.com/c/8uEbvOoz/124-as-an-instructor-i-can-post-newsfeed-articles-to-provide-relevant-resources-to-students

LOOM RECORDING:  https://www.loom.com/share/3275a8a1bcdc4100bc2d0b0ec7bb5ad7

I implemented the newsfeed end-points, model, and did some simple middle-ware.
I both found and fixed a security risk because the delete end-point did not have the auth required middle-ware.
And updated the test file for the newsfeed to reflect the new code.

Fixes #
the delete end-point did not have the auth required middle-ware.

## Type of change
Implementation of new end-points for the front-end

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by Ryan-E-Mark — APPROVED** [review]
Looks good! It will be neat to get your backend hooked up to Jonah and Jaehun's front end components and have some real CRUD functionality!

I noticed there is some work on the inbox routes and other non-newsfeed related things. Not an issue but in the future probably best to keep changes related to the branch you're working on to avoid merge conflicts or crossing over of work.

Keep up the great work Leo!

**Review by seanmp3 — APPROVED** [review]
Looks really good man! Only nitpick I can think of is that some of the model files have a oneliner return while others have it more spaced out, but even with that its all pretty readable so its probably alright.

File: __tests__/routes/newsfeed.test.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ describe('newsfeeds router endpoints', () => {
   });
 
   describe('POST /newsfeed', () => {
-    it('Should return 200 when newsfeed is added', async () => {
```
Comment (jonahjtr) [inline __tests__/routes/newsfeed.test.js:50]: great catch using the correct status number! remembering the 201 when posting something and 200 when calling something is great attention to detail

**Review by jonahjtr — APPROVED** [review]
great pull request! I'd focus on staying on "branch-specific" files for the off chance someone else breaks code and it looks like you touched it last (just a joke)
great job!

**Pull request merged**